src/layers/core/point-density-grid-layer/point-density-grid-layer.js

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {Layer} from '../../../lib';
+import GridLayer from '../grid-layer';
+
+import {pointToDensityGridData} from './grid-aggregator';
+
+const defaultCellSize = 1;
+const defaultColorRange = [
+  [255, 255, 178],
+  [254, 217, 118],
+  [254, 178, 76],
+  [253, 141, 60],
+  [240, 59, 32],
+  [189, 0, 38]
+];
+const defaultElevationRange = [0, 10];
+
+const defaultProps = {
+  cellSize: defaultCellSize,
+  colorRange: defaultColorRange,
+  elevationRange: defaultElevationRange
+};
+
+function noop() {}
+
+function _needsReProjectPoints(oldProps, props) {
+  return oldProps.cellSize !== props.cellSize;
+}
+
+function _ordinalScale(domain, range, value) {
+  const step = (domain[1] - domain[0]) / range.length;
+  const idx = Math.floor((value - domain[0]) / step);
+  const clampIdx = Math.max(Math.min(idx, range.length - 1), 0);
+
+  return range[clampIdx];
+}
+
+function _linearScale(domain, range, value) {
+
+  return (value - domain[0]) / (domain[1] - domain[0]) * (range[1] - range[0]) + range[0];
+}
+
+export default class PointDensityGridLayer extends Layer {
+  initializeState() {
+    this.state = {
+      gridOffset: {latOffset: 0.0089, lonOffset: 0.0113},
```

Comment:
Here we are still tied to a specific projection mode. Even if we need more time to generalize the calculations, it would be good if we could avoid tying the names to lat/lon.